### ISSUE TITLE
[NeatoVacuumRobot] unkown error - Can't log on to neato cloud. Please check your credentials.

### ISSUE THREAD

**Issue by DJay-X** [naofireblade/homebridge-neato#19]
Hi there..
It's probably the first time I'm experience this, but homebridge stopped because of the homebridge-neato plugin.
It could be because of the recent update from Neato and they have serve issues, but maybe there is a way that the plugin can handle this different than stopping whole homebridge from work. 

Thanks in advance for considering a fix.

```
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]: [2018-12-4 17:29:35] [NeatoVacuumRobot] unkown error
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]: [2018-12-4 17:29:35] [NeatoVacuumRobot] Can't log on to neato cloud. Please check your credentials.
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]: /usr/local/lib/node_modules/homebridge-neato/node_modules/node-botvac/lib/client.js:23
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:                         callback('unkown error');
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:                         ^
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]: TypeError: Cannot read property 'length' of undefined
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at /usr/local/lib/node_modules/homebridge-neato/index.js:43:36
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at /usr/local/lib/node_modules/homebridge-neato/index.js:60:5
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at Client.<anonymous> (/usr/local/lib/node_modules/homebridge-neato/node_modules/node-botvac/lib/client.js:23:25)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at Request._callback (/usr/local/lib/node_modules/homebridge-neato/node_modules/node-botvac/lib/api.js:42:45)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at Request.self.callback (/usr/local/lib/node_modules/homebridge-neato/node_modules/request/request.js:185:22)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at Request.emit (events.js:182:13)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at Request.<anonymous> (/usr/local/lib/node_modules/homebridge-neato/node_modules/request/request.js:1161:10)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at Request.emit (events.js:182:13)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/homebridge-neato/node_modules/request/request.js:1083:12)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at Object.onceWrapper (events.js:273:13)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at IncomingMessage.emit (events.js:187:15)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at endReadableNT (_stream_readable.js:1098:12)
Dez 04 17:29:35 raspberrypi-3BPlus homebridge[1899]:     at process.internalTickCallback (internal/process/next_tick.js:72:19)
Dez 04 17:29:35 raspberrypi-3BPlus systemd[1]: homebridge.service: Main process exited, code=exited, status=1/FAILURE
Dez 04 17:29:35 raspberrypi-3BPlus systemd[1]: homebridge.service: Unit entered failed state.
Dez 04 17:29:35 raspberrypi-3BPlus systemd[1]: homebridge.service: Failed with result 'exit-code'.
```

**Comment by ifeign** [comment]
Their whole server was down, it's working again now. However, it is frustrating that this one plugin not being able to log in breaks the rest of Homebridge.

**Comment by DJay-X** [comment]
@ifeign You shouldn't be frustrated. Well.. personally I never had this experience before. And I assume neither @naofireblade did.
So I'm confident that @naofireblade will look into this for maybe one of the next updates.

**Comment by DJay-X** [comment]
Issue never appeared again since Dec. 2018. So I close this. 
Thanks @naofireblade for your great work, sharing with us and the ongoing development. 👍🏻

**Comment by naofireblade** [comment]
Fixed in v0.6.3 so that homebridge does not crash if the neato servers are offline.
